single_tech_samples/synapseanalytics/sample3_integration_testing/tests/integrationtests/utils/pipelineutils.py

```diff
@@ -0,0 +1,222 @@
+from datetime import datetime, timedelta
+import time
+from typing import List
+from azure.synapse.artifacts import ArtifactsClient
+from azure.synapse.artifacts.models._models_py3 import (
+    RunQueryFilter,
+    RunFilterParameters,
+)
+import json
+
+
+def run_and_observe_pipeline(
+    synapse_client: ArtifactsClient,
+    pipeline_name: str,
+    params: dict = None,
+) -> str:
+    """Create a pipeline run and get the status on completion
+
+    Args:
+        synapse_client (ArtifactsClient): Synapse Connection Client
+        pipeline_name (str): Name of the Pipeline to run
+        params (dict): Parameters for the Synapse Pipeline
+
+    Returns:
+        pipeline_run_status (str): Status of the pipeline run
+    """
+    pipeline_run_id = run_pipeline(synapse_client, pipeline_name, params)
+    pipeline_run_status = observe_pipeline(synapse_client, pipeline_run_id)
+    return pipeline_run_status
+
+
+def run_pipeline(
+    synapse_client: ArtifactsClient,
+    pipeline_name: str,
+    params: dict = None,
+) -> str:
+    """Create a pipeline run and get the run id
+
+    Args:
+        synapse_client (ArtifactsClient): Synapse Connection Client
+        pipeline_name (str): Name of the Pipeline to run
+        params (dict): Parameters for the Synapse Pipeline
+
+    Returns:
+        (str): ID of the pipeline run
+    """
+    print(f'CREATING THE PIPELINE RUN for "{pipeline_name}"...')
+
+    try:
+        run_pipeline = synapse_client.pipeline.create_pipeline_run(
+            pipeline_name, parameters=params
+        )
+        pipeline_run_id = run_pipeline.run_id
+        print(f'CREATED THE PIPELINE RUN and got the run id "{pipeline_run_id}"')
+        return pipeline_run_id
+    except Exception as e:
+        print(e)
+        raise
+
+
+def observe_pipeline(
```

Comment:
should there be a default and possibly override able parameter for the integration test suite to timeout and error out?

This is very tightly dependent on the test scenario, and the synapse resources, but having some timeout may be better than not having any

Reply:
The observe_pipeline method is nearly identical to the e2e sample method aside from a try-catch, so same comment as above. cc @devlace :)